pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/embeddings/AbstractHuggingFaceEmbeddingService.java

```diff
@@ -95,7 +97,11 @@ public static class HuggingFaceConfig {
   public AbstractHuggingFaceEmbeddingService(HuggingFaceConfig conf)
       throws IOException, ModelNotFoundException, MalformedModelException, IllegalAccessException {
     Objects.requireNonNull(conf);
-    Objects.requireNonNull(conf.modelUrl);
+
+    if (conf.modelUrl == null) {
+      conf.modelUrl = DLJ_BASE_URL + conf.modelName;
```

Comment:
FYI model name and the archive name aren't necessary the same but this should work in basic cases. 
Some archives can include multiple models etc. 
http://djl.ai/docs/load_model.html#customize-modelname

In this case it makes sense to set conf.modelName to null to avoid using it the way optModelName is used.

Reply:
Overall this logic should not be here. 
It belongs in TransformFunction where the service config is built from transform's config:
```java
          case LOCAL:
...
            AbstractHuggingFaceEmbeddingService.HuggingFaceConfig.HuggingFaceConfigBuilder builder =
                AbstractHuggingFaceEmbeddingService.HuggingFaceConfig.builder()
...
```

Reply:
I have moved this code to TransformFunction.

In my testing I need to set both modelUrl and modelName, so I am not setting modeName to null